assets/js/autocomplete.ts

```diff
@@ -6,52 +6,67 @@ import { LocalAutocompleter } from './utils/local-autocompleter';
 import { handleError } from './utils/requests';
 import { getTermContexts } from './match_query';
 import store from './utils/store';
+import { TermContext } from './query/lex.ts';
+import { $, $$, makeEl, removeEl } from './utils/dom.ts';
 
-const cache = {};
-/** @type {HTMLInputElement} */
-let inputField,
-  /** @type {string} */
-  originalTerm,
-  /** @type {string} */
-  originalQuery,
-  /** @type {TermContext} */
-  selectedTerm;
+type TermSuggestion = {
+  label: string;
+  value: string;
+};
+
+const cachedSuggestions: Record<string, TermSuggestion[]> = {};
+let inputField: HTMLInputElement | null = null,
+  originalTerm: string | undefined,
+  originalQuery: string | undefined,
+  selectedTerm: TermContext | null = null;
 
 function removeParent() {
-  const parent = document.querySelector('.autocomplete');
-  if (parent) parent.parentNode.removeChild(parent);
+  const parent = $<HTMLElement>('.autocomplete');
+  if (parent) removeEl(parent);
 }
 
 function removeSelected() {
-  const selected = document.querySelector('.autocomplete__item--selected');
+  const selected = $<HTMLElement>('.autocomplete__item--selected');
   if (selected) selected.classList.remove('autocomplete__item--selected');
 }
 
-function isSearchField() {
-  return inputField && inputField.dataset.acMode === 'search';
+function isSearchField(targetInput: HTMLElement) {
```

Comment:
```suggestion
function isSearchField(targetInput: HTMLElement): boolean {
```

Reply:
Oh, just need to explicitly show the type? Alright.